src/fe-common/core/fe-common-core.c

```diff
@@ -461,26 +461,45 @@ void fe_common_core_finish_init(void)
 
 gboolean strarray_find_dest(char **array, const TEXT_DEST_REC *dest)
 {
+	int channel_type = module_get_uniq_id_str("WINDOW ITEM TYPE", "CHANNEL");
+	int query_type = module_get_uniq_id_str("WINDOW ITEM TYPE", "QUERY");
+	char **tmp;
+	char *slash = NULL;
+
 	g_return_val_if_fail(array != NULL, FALSE);
+	g_return_val_if_fail(dest != NULL, FALSE);
+	g_return_val_if_fail(dest->window != NULL, FALSE);
+	g_return_val_if_fail(dest->target != NULL, FALSE);
 
-	if (strarray_find(array, "*") != -1)
-		return TRUE;
+	WI_ITEM_REC *item = window_item_find_window(dest->window, dest->server, dest->target);
+	g_return_val_if_fail(item != NULL, FALSE);
 
-	if (strarray_find(array, dest->target) != -1)
-		return TRUE;
+	for (tmp = array; *tmp != NULL; tmp++) {
+		char *str = *tmp;
+		if (*str == '\0') {
+			continue;
+		}
+		if (dest->server_tag != NULL && (slash = strchr(str, '/'))) {
+			*slash = '\0';
+			if (g_ascii_strcasecmp(str, dest->server_tag) == 0) {
+				str = slash + 1;
+			}
+			*slash = '/';
+		}
 
-	if (dest->server_tag != NULL) {
-		char *tagtarget = g_strdup_printf("%s/%s", dest->server_tag, "*");
-		int ret = strarray_find(array, tagtarget);
-		g_free(tagtarget);
-		if (ret != -1)
+		if (g_strcmp0(str, "") == 0 || g_strcmp0(str, "::all") == 0) {
 			return TRUE;
-
-		tagtarget = g_strdup_printf("%s/%s", dest->server_tag, dest->target);
-		ret = strarray_find(array, tagtarget);
-		g_free(tagtarget);
-		if (ret != -1)
+		} else if (g_ascii_strcasecmp(str, dest->target) == 0) {
 			return TRUE;
+		} else if (item->type == query_type &&
+			(dest->target[0] == '=' ?
+			g_strcmp0(str, "::dccqueries") == 0 : g_strcmp0(str, "::queries") == 0)) {
```

Comment:
You can shorten the code by swapping the ternary operator and the second argument to `g_strcmp0`:
```c
g_strcmp0(str, (dest->target[0] == '=') ? "::dccqueries: : "::queries")
```